full/flux_exchange.F90

```diff
@@ -516,34 +503,13 @@ module flux_exchange_mod
 !REGRID: physically distinct grids, via xgrid
 !REDIST: same grid, transfer in index space only
 !DIRECT: same grid, same decomp, direct copy
-  use atmos_model_mod,    only: atmos_data_type, land_ice_atmos_boundary_type
-  use ocean_model_mod,    only: ocean_public_type, ice_ocean_boundary_type
-  use ocean_model_mod,    only: ocean_state_type
-  use ice_model_mod,      only: ice_data_type, land_ice_boundary_type, &
-                                ocean_ice_boundary_type, atmos_ice_boundary_type, Ice_stock_pe
-  use land_model_mod,     only: land_data_type, atmos_land_boundary_type
-  use xgrid_mod,          only: get_ocean_model_area_elements
-  use  time_manager_mod,  only: time_type
-  use sat_vapor_pres_mod, only: sat_vapor_pres_init
-  use      constants_mod, only: rdgas, rvgas, cp_air, stefan, WTMAIR, HLV, HLF, Radius, PI, CP_OCEAN, &
-                                WTMCO2, WTMC
-!Balaji
-!utilities stuff into use fms_mod
-  use fms2_io_mod,                only: open_file, close_file, FmsNetcdfDomainFile_t, FmsNetcdfFile_t
-  use fms2_io_mod,                only: get_variable_size, get_variable_dimension_names, variable_exists
-  use fms2_io_mod,                only: read_data, register_field, register_axis
-  use fms_mod,                    only: clock_flag_default, check_nml_error, error_mesg
-  use fms_mod,                    only:  write_version_number
-  use data_override_mod,          only: data_override
-  use coupler_types_mod,          only: coupler_1d_bc_type
-  use atmos_ocean_fluxes_mod,     only: atmos_ocean_fluxes_init, atmos_ocean_type_fluxes_init
-  use atmos_ocean_fluxes_calc_mod, only: atmos_ocean_fluxes_calc
-  use ocean_model_mod,            only: ocean_model_init_sfc, ocean_model_flux_init
-  use atmos_tracer_driver_mod,    only: atmos_tracer_flux_init
-  use stock_constants_mod,        only: NELEMS, ISTOCK_WATER, ISTOCK_HEAT, ISTOCK_SALT
-  use stock_constants_mod,        only: ISTOCK_SIDE, ISTOCK_TOP, ISTOCK_BOTTOM , STOCK_UNITS, STOCK_NAMES
-  use stock_constants_mod,        only: stocks_file, stocks_report, stocks_report_init
-  use stock_constants_mod,        only: Atm_stock, Ocn_stock, Lnd_stock, Ice_stock
+
+  use FMS, tfreeze_fms => tfreeze, version_fms => version
+
+  !! fms_io not in global module(unsupported)
+  use fms_io_mod, only:  write_version_number
```

Comment:
Once write_version_number is moved to fms_mod and is publicized via 'use FMS', this will need to be modified.